lib/sles4sap/sdaf_library.pm

```diff
@@ -513,14 +720,124 @@ sub prepare_sdaf_repo {
     assert_script_run("mkdir -p $_") foreach @create_workspace_dirs;
 }
 
-=head2 cleanup_sdaf_files
+=head2 generate_resource_group_name
+
+    generate_resource_group_name(deployment_type=>$deployment_type);
+
+B<$deployment_type>: Type of the deployment (workload_zone, sap_system, library... etc)
+
+Returns name of the resource group for the deployment type specified by B<$deployment_type> .
+Resource group pattern: I<SDAF-OpenQA-[deployment type]-[deployment id]-[OpenQA job id]>
+
+=cut
+
+sub generate_resource_group_name {
+    my (%args) = @_;
+    my @supported_types = ('workload_zone', 'sap_system', 'library', 'deployer');
+    croak "Unsupported deployment type: $args{deployment_type}\nCurrently supported ones are: @supported_types" unless
+      grep(/^$args{deployment_type}$/, @supported_types);
+    my $job_id = get_current_job_id();
+    my $resource_group = join('-', 'SDAF', 'OpenQA', $args{deployment_type}, $job_id);
```

Comment:
```suggestion
    return  join('-', 'SDAF', 'OpenQA', $args{deployment_type}, $job_id);
```

Reply:
Fixed. Wasn't sure about readability if it just returns some joined string.